bamboo/abort-if-not-pr-or-master.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```

Comment:
These 'abort' scripts are all workarounds to allow bamboo to halt (but not fail) jobs that aren't needed in the current execution configuration. 

There's a couple of other ways to do this (e.g. export -f and the like), but calling the functions inline  felt both more readable and allows us to not need to return updated environment settings, should we need to do so).